modules/service/src/main/scala/lucuma/odb/service/TimeAccountingService.scala

```diff
@@ -375,13 +385,47 @@ object TimeAccountingService {
           e.c_step_id
         FROM
           t_execution_event e
-        LEFT JOIN t_step_record s ON s.c_step_id = e.c_step_id
+        LEFT JOIN t_step_record s USING (c_step_id)
         WHERE
           e.c_visit_id = $visit_id
         ORDER BY
           e.c_received
       """.query(codec.event)
 
+    val SelectChargeableOverlap: Query[Visit.Id, (Timestamp, Observation.Id)] =
+      sql"""
+        WITH v AS (
+          SELECT
+            c_visit_id,
+            c_site,
+            c_start,
+            c_end
+          FROM t_visit
+          WHERE c_visit_id = $visit_id
+        )
+        SELECT
+          v2.c_start,
+          v2.c_observation_id
+        FROM
+          t_visit v2, v
+        WHERE
+          v2.c_visit_id != v.c_visit_id AND
+          v2.c_site = v.c_site          AND
+          v2.c_chargeable = true        AND
+          v2.c_start < v.c_end          AND
+          (
+            -- The usual case if v2 overlaps
+            v2.c_start > v.c_start                                    OR
+
+            -- The edge case where the start times are the same on two visits.
+            -- Here we just pick the one with the newer id to be charged.
+            (v2.c_start = v.c_start AND v2.c_visit_id > v.c_visit_id)
```

Comment:
This will never happen, of course, but if it did we'd double charge these observations without this bit.  It just picks one at random (the one whose visit was last created) to charge and the other one is free.